src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -106,9 +108,17 @@ public String getScript() {
         if (script.isEmpty()) {
             listener.getLogger().println("Warning: was asked to run an empty script");
         }
+        OsType os = ws.act(new getOsType());
+        String scriptEncodingCharset = "UTF-8";
+        if(os == OsType.ZOS) {
+            if(SYSTEM_DEFAULT_CHARSET.equals(getCharset())) {
+            // Setting default charset to IBM z/OS default EBCDIC charset on z/OS if no encoding specified on sh step
```

Comment:
Because we can't hardcode normally...  EBCDIC is a set of codepages where symbols mapping differs. Worst thing that in EBCDIC that affects such chars like '! [ ]'  Default code page is  IBM-1047. but z/OS system programmer may set any other, like IBM-1025.  And if for example i'll hardcode IBM-1047 code page, but system is configured in IBM-1025 i will get bad chars in a script itself and it will fail. 
As example 
#!/bin/bash  will turn to smth like #|/bin/bash
'[ -f myfile ];'   will turn to smth like 'p -f myfille y;'

So the logic implemented is following (on z/OS only)
1/ If encoding parameter is set on sh encoding:  step - use it for transcode script, script output and script exit status)
2/ If no encoding specified on sh encoding: step -  go to z/OS and ask what codepage is set and apply it for transcoding

Reply:
Because we can't hardcode normally... EBCDIC is a set of codepages where symbols mapping differs. Worst thing that in EBCDIC that affects such chars like '! [ ]' Default code page is IBM-1047. but z/OS system programmer may set any other, like IBM-1025. And if for example i'll hardcode IBM-1047 code page, but system is configured in IBM-1025 i will get bad chars in a script itself and it will fail.
As example
#!/bin/bash will turn to smth like #|/bin/bash
'[ -f myfile ];' will turn to smth like 'p -f myfille y;'

So the logic implemented is following (on z/OS only)
1/ If encoding parameter is set on sh encoding: step - use it for transcode script, script output and script exit status)
2/ If no encoding specified on sh encoding: step - go to z/OS and ask what codepage is set and apply it for transcoding

Reply:
I just wanted to give an option to user to manually set, encoding of the script on z/OS platform regardless of what is set on a system. But i can revert it back to Jenkins logic where sh encoding: determines only an output encoding, no problems here.